### DISCUSSION TITLE
low bit depth dithered PNG with fixed or no palette

### DISCUSSION THREAD

**Discussion by pepijndevos** [libvips/libvips#3008] [Ideas]
I am trying to make an 1 bit per pixel PNG image, but I want the colors to be pure black and white. When setting bit depth to 1, it either uses a palette that may contain grayscales, which looks really bad when tiled, or it does thresholding without dithering, which is also suboptimal.

![image](https://user-images.githubusercontent.com/168609/182885573-f68071f8-1cc1-4197-83d3-604ec00dc362.png)

So what I'd like to be able to do is either specify a fixed palette, or do dithering without using a palette at all, but I've been told by the sharp people that libvips can't do this.

https://github.com/lovell/sharp/issues/3322

**Comment by jcupitt** [comment]
Hi @pepijndevos,

You could do the thresholding yourself, eg.:

```python
(image.colourspace("b-w") > 128).write_to_file("x.png", bitdepth=1)
```

But that won't dither.

Dithering will break on tile boundaries if you save a tile at a time -- we'd need to add a dither operation to fix that, I think.

**Reply by pepijndevos** [reply]
The tiles are generated with some padding to avoid edge effects, but yes a dither option would be needed that allows you to either specify the palette, or work without a palette.

**Reply by jcupitt** [reply]
OK, let's make this into an enhancement issue so we don't forget.

**Reply by jcupitt** [reply]
https://github.com/libvips/libvips/issues/3010 since github seems unable to link these automatically
